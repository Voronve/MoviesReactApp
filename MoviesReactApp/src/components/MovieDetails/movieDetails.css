.movieDetailsWrapper {
    padding: 50px;
    display: flex;
    font-family: 'Montserrat', sans-serif;;
}

.movieDetailsWrapper .movieDataWrapper .titleAndRating {
    display: flex;
}

.movieDetailsWrapper .movieDataWrapper .yearAndDuration {
    display: flex;
}

.movieDataWrapper {
    margin: 0 40px;
}

.movieDataWrapper .titleAndRating {
    display: flex;
    color: white;
    align-items: center;
}

.movieDataWrapper .titleAndRating .title{
   font-size: 50px;
   text-transform: uppercase;
}

.movieDataWrapper .titleAndRating .vote_average{
    margin-left: 10px;
    align-self:center;
    text-align: center;
    height: 50px;
    width: 50px;
    border: 2px solid white;
    border-radius: 50%;
    line-height: 50px;
    font-size: 20px;
}

.movieDataWrapper .ganres, .movieDataWrapper .overview {
    color: #8b8888;
}

.movieDataWrapper .yearAndDuration {
    color: red;
    margin: 30px 0;
    font-size: 25px;
}

.movieDataWrapper .runtime {
    margin-left: 50px;;
}
