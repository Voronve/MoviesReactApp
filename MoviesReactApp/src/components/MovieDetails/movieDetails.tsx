import './movieDetails.css';
import { MovieInfo } from '../MovieListPage/movieListPage';

interface MovieDetailsProps {
    /** All movie data*/
    movieData: MovieInfo | null;
}

/** Movie details section*/
export function MovieDetails({ movieData }: MovieDetailsProps ) {
    if(!movieData || !Object.keys(movieData).length) {

        return null;
    }

    const {poster_path, title, release_date, vote_average, genres, runtime, overview} = movieData;

    return (
        <div className="movieDetailsWrapper">
            <div className='poster'>
                <img src={`${poster_path}`} alt={title} width="322px"/>
            </div>
            <div className='movieDataWrapper'>
                <div className="titleAndRating">
                    <div className="title">{title}</div>
                    <div className="vote_average">{vote_average}</div>
                </div>
                <div className="ganres">{genres.join()}</div>
                <div className="yearAndDuration">
                    <div className="year">{release_date}</div>
                    <div className="runtime">{runtime}</div>
                </div>
                <div className="overview">{overview}</div>
            </div>
        </div>
    );
}