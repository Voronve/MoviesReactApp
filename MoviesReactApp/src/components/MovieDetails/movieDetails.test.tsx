import { MovieDetails } from './movieDetails';
import { render } from '@testing-library/react';

describe("Testing MovieDetails component", () => {

    test("Testing movie details rendering", () => {

        const movieInfo = {
            poster_path: 'testImage',
            title:'testImage',
            release_date: 1987,
            genres: ["DRAMA", "BIOPIC"],
            vote_average: 9.9,
            runtime: 'infinity',
            overview: 'lorem ipsum'
        }

        const { asFragment } = render(<MovieDetails movieData = {movieInfo}/>);

        expect(asFragment()).toMatchSnapshot();
    });
});