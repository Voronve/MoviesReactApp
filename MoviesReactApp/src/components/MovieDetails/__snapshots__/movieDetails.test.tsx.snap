// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieDetails component Testing movie details rendering 1`] = `
<DocumentFragment>
  <div
    class="movieDetailsWrapper"
  >
    <div
      class="poster"
    >
      <img
        alt="testImage"
        src="testImage"
        width="322px"
      />
    </div>
    <div
      class="movieDataWrapper"
    >
      <div
        class="titleAndRating"
      >
        <div
          class="title"
        >
          testImage
        </div>
        <div
          class="vote_average"
        >
          9.9
        </div>
      </div>
      <div
        class="ganres"
      >
        DRAMA,BIOPIC
      </div>
      <div
        class="yearAndDuration"
      >
        <div
          class="year"
        >
          1987
        </div>
        <div
          class="runtime"
        >
          infinity
        </div>
      </div>
      <div
        class="overview"
      >
        lorem ipsum
      </div>
    </div>
  </div>
</DocumentFragment>
`;
